File: db4oj.tests/src/com/db4o/db4ounit/jre5/collections/typehandler/ListTypeHandlerTestUnitBase.java
package com.db4o.db4ounit.jre5.collections.typehandler;

import java.util.*;

import com.db4o.typehandlers.*;

/**
 */
@decaf.Ignore(unlessCompatible=decaf.Platform.JDK15)
public class ListTypeHandlerTestUnitBase extends TypeHandlerTestUnitBase {
	
	protected AbstractItemFactory itemFactory() {
		return (AbstractItemFactory) ListTypeHandlerTestVariables.LIST_IMPLEMENTATION.value();
	}
	
	protected TypeHandler4 typeHandler() {
	    return (TypeHandler4) ListTypeHandlerTestVariables.LIST_TYPEHANDER.value();
	}
	
	protected void fillItem(Object item) {
		fillListItem(item);
	}

	protected void assertContent(Object item) {
		assertListContent(item);
	}

	protected void assertPlainContent(Object item) {
		assertPlainListContent((List) item);
	}

	protected ListTypeHandlerTestElementsSpec elementsSpec() {
		return (ListTypeHandlerTestElementsSpec) ListTypeHandlerTestVariables.ELEMENTS_SPEC.value();
	}    


}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Jre5/Collections/Typehandler/ListTypeHandlerTestUnitBase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System.Collections;
using Db4objects.Db4o.Tests.Jre5.Collections.Typehandler;
using Db4objects.Db4o.Typehandlers;

namespace Db4objects.Db4o.Tests.Jre5.Collections.Typehandler
{
	public class ListTypeHandlerTestUnitBase : TypeHandlerTestUnitBase
	{
		protected override AbstractItemFactory ItemFactory()
		{
			return (AbstractItemFactory)ListTypeHandlerTestVariables.ListImplementation.Value;
		}

		protected override ITypeHandler4 TypeHandler()
		{
			return (ITypeHandler4)ListTypeHandlerTestVariables.ListTypehander.Value;
		}

		protected override void FillItem(object item)
		{
			FillListItem(item);
		}

		protected override void AssertContent(object item)
		{
			AssertListContent(item);
		}

		protected override void AssertPlainContent(object item)
		{
			AssertPlainListContent((IList)item);
		}

		protected override ListTypeHandlerTestElementsSpec ElementsSpec()
		{
			return (ListTypeHandlerTestElementsSpec)ListTypeHandlerTestVariables.ElementsSpec
				.Value;
		}
	}
}
